code/game/turfs/turf.dm

```diff
@@ -252,6 +252,16 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
 			return 1
 	return 0
 
+/// Returns TRUE if the turf is occupied by a big object already (i.e. computer)
+/proc/is_occupied_turf(turf/T, exclude_mobs=FALSE)
+	if(is_blocked_turf(T, exclude_mobs)) // check walls first
+		return TRUE
+	for(var/obj/O in T)
+		if(O.density) // something big occupied the turf
+			return TRUE
+	return FALSE
```

Comment:
Yeah isn't this new proc completely useless since `is_blocked_turf` already does a density check for everything on/in the turf?

Reply:
As far as I interpreted code well, the proc for the turf doesn't care object desity which why it is placed onto glass or whacky places, and I didn't wanted to reveiw the current full usage of the proc because adding a new logic to the proc can possibly cause uxpectable fails (many return TRUE). So I thought having a new proc with additional logic is safer to solve this issue

Reply:
You are correct that the current logic for the, in this case the exploration nuke, placement isn't working well. The thing i mean though is that this new proc does what `is_blocked_turf` also does (the density checking of things on/in the turf)

So my guess is that replacing the usage of `is_occupied_turf` with `is_blocked_turf` in ruin_generator.dm that you added would be enough

Reply:
That seems to be the solution, yes

Reply:
Is blocked turf checks density, its the proc above and is exactly rhe same, just using atom instead of obj

Reply:
ah you're right. then it seems `is_blocked_turf` might work properly? or sanity check 100 wasn't sufficient... I will change this PR a bit.

Reply:
okay, it seems I was confused with something because of the existence of
```
			if(locate(/obj/structure) in T)
				continue
```
now it will solve the issue, I believe.